pkg/kubelet/pod_manager.go

```diff
@@ -114,6 +115,17 @@ func (pm *basicPodManager) UpdatePods(u PodUpdate, podSyncTypes map[types.UID]me
 		}
 		allPods := applyUpdates(u.Pods, pm.getAllPods())
 		pm.setPods(allPods)
+
+	case cache.ADD:
+		glog.V(3).Infof("Add: Containers changed")
+		for i := range u.Pods {
+			podSyncTypes[u.Pods[i].UID] = metrics.SyncPodCreate
+		}
+		// TODO: Something about thread safety needs thinking about here.
+		allPods := u.Pods
+		allPods = append(allPods, pm.getAllPods()...)
+		pm.setPods(allPods)
+
 	default:
 		panic("syncLoop does not support incremental changes")
```

Comment:
Is this still true?

I'm not familiar with the reason to not accept incremental changes. Maybe others can chime in.